qrenderdoc/Windows/ShaderMessageViewer.cpp

```diff
@@ -778,9 +781,15 @@ void ShaderMessageViewer::refreshMessages()
       else
         location += lit(", Prim %1").arg(msg.location.pixel.primitive);
 
+      // only show the view if the draw has multiview enabled
+      if(m_Multiview)
+      {
+        location += lit(", View %1").arg(msg.location.pixel.view);
+      }
+
       if(m_Multisampled)
       {
-        if(msg.location.pixel.sample == ~0U)
+        if(msg.location.pixel.sample == (~0U >> 16))
```

Comment:
This is a leaky abstraction, the UI level code shouldn't have to worry about bitpacking like this otherwise this code would need to be replicated in anyone using the replay API (which is currently documented to return the full `~0U`).

Instead when decoding the bitpacking of sample/view you should detect if the sample is `~0U >> 16` and sign-extend by setting the upper 16 bits.

Reply:
Moved to 
```
          if(msg.location.pixel.sample == (~0U >> 16U))
          {
            msg.location.pixel.sample = ~0U;
          }
```
in vk_shader_feedback.cpp instead.